File: unit-3/sprint-2/evaluation/firstapp/app/src/main/java/com/example/firstapp/ItemAdapter.java
package com.example.firstapp;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class ItemAdapter extends RecyclerView.Adapter<itemViewHolder> {

    private FragmentListner fragmentListner;
    private List<ResponseModel> itemList;

    public ItemAdapter(FragmentListner fragmentListner, List<ResponseModel> itemList) {
        this.fragmentListner = fragmentListner;
        this.itemList = itemList;
    }

    @NonNull
    @Override
    public itemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout,parent,false);
        return new itemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull itemViewHolder holder, int position) {
        ResponseModel responseModel = itemList.get(position);
        holder.setData(responseModel);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String title = responseModel.getTitle();
                String subTitle= responseModel.getSubTitle();
                String Imageurl = responseModel.getImage();
                Bundle bundle = new Bundle();
                bundle.putString("title",title);
                bundle.putString("subtitle",subTitle);
                bundle.putString("Imageurl",Imageurl);
                fragmentListner.onItemClicked(bundle);
            }
        });
    }

    @Override
    public int getItemCount() {
        return itemList.size();
    }
}

File: unit-4/sprint-2/evaluation/unit4_codingeval_1/app/src/main/java/com/example/unit4_codingeval_1/recyclerview/ItemAdapter.kt
package com.example.unit4_codingeval_1.recyclerview

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.unit4_codingeval_1.R

class ItemAdapter(val context: Context, val list: MutableList<ModelClass>) :
    RecyclerView.Adapter<ItemViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        return ItemViewHolder(
            LayoutInflater.from(context).inflate(R.layout.item_layout, parent, false)
        )
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        holder.setData(list[position])
    }

    override fun getItemCount(): Int {
        return list.size
    }
}